apps/openmw/mwmechanics/actors.cpp

```diff
@@ -422,6 +423,26 @@ namespace MWMechanics
             MWBase::Environment::get().getDialogueManager()->say(actor, "idle");
     }
 
+    void Actors::updateMovementSpeed(const MWWorld::Ptr& actor)
+    {
+        float previousSpeedFactor = actor.getClass().getMovementSettings(actor).mSpeedFactor;
+        float newSpeedFactor = 1.f;
+
+        CreatureStats &stats = actor.getClass().getCreatureStats(actor);
+        MWMechanics::AiSequence& seq = stats.getAiSequence();
+
+        if (!seq.isEmpty() && seq.getActivePackage()->useVariableSpeed())
+        {
+            osg::Vec3f targetPos = seq.getActivePackage()->getDestination();
+            osg::Vec3f actorPos = actor.getRefData().getPosition().asVec3();
+            float distance = (targetPos - actorPos).length();
```

Comment:
Maybe compare length2() result with DECELERATE_DISTANCE * DECELERATE_DISTANCE to avoid an unnecessary square root extraction for every actor every frame.

previousSpeedFactor should never be set to a value above 1, so a std::min can be removed.

Works good otherwise, apart from a few quirks not specific to the addition.